File: app/schemas/blog.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class BlogAuthor(BaseModel):
    id: int
    name: str
    avatar_url: Optional[str] = None
    bio: Optional[str] = None

    class Config:
        from_attributes = True

class BlogPostBase(BaseModel):
    title: str
    content: str
    description: str
    image_url: Optional[str] = None
    tags: List[str] = []
    published: bool = False

class BlogPostCreate(BlogPostBase):
    pass

class BlogPostUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    description: Optional[str] = None
    image_url: Optional[str] = None
    tags: Optional[List[str]] = None
    published: Optional[bool] = None

class BlogPost(BlogPostBase):
    id: int
    slug: str
    created_at: datetime
    updated_at: datetime
    reading_time: str
    author: BlogAuthor

    class Config:
        from_attributes = True

File: frontend/src/types/blog.ts
export interface BlogAuthor {
  id: number;
  name: string;
  avatar_url?: string;
  bio?: string;
}

export interface BlogPost {
  id: number;
  title: string;
  slug: string;
  content: string;
  description: string;
  created_at: string;
  updated_at: string;
  reading_time: string;
  image_url?: string;
  tags: string[];
  author: BlogAuthor;
  published: boolean;
  in_menu: boolean;
}

export interface BlogPostCreate {
  title: string;
  content: string;
  description: string;
  image_url?: string;
  tags: string[];
  published?: boolean;
  in_menu?: boolean;
}

export interface BlogPostUpdate extends Partial<BlogPostCreate> {}
